Update Analog Input HW and SW articles for Systemcore (#3079)
wpilibsuite/frc-docs: source/docs/hardware/sensors/analog-inputs-hardware.rst

.. note:: This section covers analog input hardware.  For a software guide to analog inputs, see :ref:`docs/software/hardware-apis/sensors/analog-inputs-software:Analog Inputs - Software`.

An [analog signal](https://en.wikipedia.org/wiki/Analog_signal) is a signal whose value can lie anywhere in a continuous interval.  This lies in stark contrast to a :doc:`digital signal <digital-inputs-hardware>`, which can take only one of several discrete values.  The Systemcore's analog input ports allow the measurement of analog signals with values from 0V to 3.3V.

.. important:: Systemcore Analog Inputs are 3.3 volts, whereas the roboRIO was 5 volts.

In practice, there is no way to measure a "true" analog signal with a digital device such as a computer (like the Systemcore).  Accordingly, the analog inputs are actually measured as a 12-bit digital signal - however, this is quite a high resolution [1]_.

> Analog inputs are typically (but not always!) used for sensors whose measurements vary continuously over a range, such as :doc:`potentiometers <analog-potentiometers-hardware>`, as they can communicate by outputting a voltage proportional to their measurements.

Reviewer: ```suggestion
Analog inputs are typically used with sensors that produce continuously varying measurements. :doc:`Potentiometers <analog-potentiometers-hardware>` are a good example, as they can communicate by outputting a voltage proportional to their measurements.
```

.. image:: images/systemcore/systemcore-smartio.png
   :alt: The Systemcore, with the location of the Smart I/O ports highlighted.

The Systemcore has 6 Smart I/O ports (numbered 0-5), as seen in the image above.  Each port has three pins - signal, power ("3.3V"), and ground ("gnd").  The "power" and "ground" pins are used to power the peripheral sensors that connect to the analog input ports - there is a constant 3.3V potential difference between the "power" and the "ground" pins [2]_.  The signal pin is the pin on which the signal is actually measured.

### Connecting a sensor to a single analog input port

.. note:: Some sensors (such as :doc:`potentiometers <analog-potentiometers-hardware>`) may have interchangeable power and ground connections.

Most analog sensors will have three wires - signal, power, and ground - corresponding precisely to the three pins of the Smart  I/O ports. They should be connected accordingly.

.. image:: images/accelerometers-hardware/adxl193-single-axis-accelerometer-to-roborio.svg
   :alt: Hooking a ADXL193 single axis accelerometer to an analog input on the roboRIO.

> .. todo:: Update above image to use the Systemcore Smart I/O ports.

Reviewer: Do we want to hold this article till we get these images?
